app/services/tag_archiver.rb

```diff
@@ -9,7 +9,9 @@ def initialize(tag, successor)
   end
 
   def archive
```

Comment:
I was trying to reduce diff in the follow up PRs. I'll be removing the first guard clause and rewording second's message. When the follow-up work is done this method should look something like this:
```
def archive
  raise "Can't archive tag with active subtopics" if tag.has_active_children?

  Tag.transaction do
     # do the thing
  end
end
```
So unless you feel strongly about it, I'll keep it as it is – I think it will make it easier to review the follow up PRs

Reply:
Sure that's fine. But I wasn't aware that we need to extend the functionality to archive level one mainstream browse? Is that a new request?

Reply:
Actually, you are right. We don't need it. I thought that it may be a bit of a tech debt not to have it given:
- both Mainstream Browse Pages and Topics inherit from Tag
- the TagArchiver has class based conditionals

Definitely a nice to have. Let's see what the Publishing team says